contracts/DomainValidator.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+library DomainValidator {
+    function stripProtocol(string memory domainName) internal pure returns (string memory) {
```

Comment:
Why not just to trim everything after the last `/` (or the first from the end)?  It even plays better as not only http and https schemas are possible. Again, always search for simpler solutions.   -1 point